pycbc/events/stat.py

```diff
@@ -2084,6 +2069,61 @@ def lognoiserate(self, trigs):
         return logr_n
 
 
+class DQExpFitFgBgKDEStatistic(DQExpFitFgBgNormStatistic):
+    """
+    The ExpFitFgBgKDEStatistic with DQ-based reranking.
+
+    This is the same as the DQExpFitFgBgNormStatistic except the signal
+    rate is adjusted according to the KDE statistic files
+    """
+
+    def __init__(self, sngl_ranking, files=None, ifos=None, **kwargs):
+        """
+        Parameters
+        ----------
+        sngl_ranking: str
+            The name of the ranking to use for the single-detector triggers.
+        files: list of strs, needed here
+            A list containing the filenames of hdf format files used to help
+            construct the coincident statistics. The files must have a 'stat'
+            attribute which is used to associate them with the appropriate
+            statistic class.
+        ifos: list of strs, not used here
+            The list of detector names
+        """
+        DQExpFitFgBgNormStatistic.__init__(self, sngl_ranking, files=files,
+                                           ifos=ifos, **kwargs)
+        self.find_kdes()
+
+    def find_kdes(self):
+        """
+        Inherited, see docstring for ExpFitFgBgKDEStatistic.find_kdes
+        """
+        ExpFitFgBgKDEStatistic.find_kdes(self)
+
+    def assign_kdes(self, kname):
```

Comment:
is a redefinition needed here, as ExpFitFgBgKDEStatistic.find_kdes calls the method defined within that class?

Reply:
I've checked through testing, and assign_kdes needs explicit inheritance, but find_kdes can be run directly

Reply:
(`AttributeError: 'DQExpFitFgBgKDEStatistic' object has no attribute 'assign_kdes'`)

Reply:
Ah, as find_kdes calls `self.assign_kdes` and 'self' is still this class instance. 
Maybe your rewrite of the KDE init method was over-complex, it could just do find_kdes and then assign_kdes without having one call the other.